File: Vidly/Customs/Extensions/Models/QueryObject.cs
﻿using System;
using System.Collections.Generic;

namespace Vidly.Customs.Extensions.Models
{
  public class QueryObject
  {
    
    public string SortBy { get; set; } = "asc";
    public string OrderBy { get; set; } = "Id";
    public int Page { get; set; } = 1;
    public int PageSize { get; set; } = 10;
    public string SearchBy { get; set; } = "";
    public string Search { get; set; } = "";
    public string Comparison { get; set; } = "contains";
    public string Fields { get; set; } = "";
  }

  public class DateRange
  {
    public DateTime DateFrom { get; set; } = DateTime.Now;
    public DateTime DateTo { get; set; } = DateTime.Now;
  }

}

File: Vidly/Scripts/Customs/Models/queryObject.js

export class QueryObject {
    constructor(page, pageSize, fields, search, searchBy, comparison, orderBy, sortBy) {
        this.page = page;
        this.pageSize = pageSize;
        this.searchBy = searchBy;
        this.search = search;
        this.comparison = comparison;
        this.orderBy = orderBy;
        this.sortBy = sortBy;
        this.fields = fields;
    }
}
